src/app/qgisapp.cpp

```diff
@@ -7504,7 +7505,7 @@ QString QgisApp::saveAsRasterFile( QgsRasterLayer *rasterLayer, const bool defau
     if ( d.outputCrs() != rasterLayer->crs() )
     {
       QgsRasterProjector *projector = new QgsRasterProjector;
-      projector->setCrs( rasterLayer->crs(), d.outputCrs() );
+      projector->setCrs( rasterLayer->crs(), d.outputCrs(), rasterLayer->dataProvider()->transformContext() );
```

Comment:
Here I'd take it from the project instance again -- just on the odd case that the data provider is out of sync (and seeing that we can safely use the project instance here)